src/main/scala/clocks/ResetWrangler.scala

```diff
@@ -0,0 +1,45 @@
+// See LICENSE for license details.
+package freechips.rocketchip.clocks
+
+import chisel3._
+import chisel3.util._
+import chisel3.experimental.{withClockAndReset}
+import freechips.rocketchip.config._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.util._
+
+class ResetWrangler(debounceNs: Double = 100000)(implicit p: Parameters) extends LazyModule
```

Comment:
Can we at least put a one line comment description of what this does....

Reply:
What I mean is, doesn't this take an async reset and turn it into a sync reset?

Reply:
I just copied this adapter from FPGA shells, but I believe you are correct. None of these have had the notion of the new chisel3.AsyncReset feature added to them, but it is a good suggestion to incorporate.

Reply:
I'm going to do it as a follow-on PR though (possibly after thanksgiving).

Reply:
what specificly does the ResetWrangler do?